### ISSUE TITLE
Site logo: set square aspect ratio by default with opt out for themes

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#63382]
To make the Site Logo block a bit more opinionated, let's consider setting the default option to square/1:1 with the option for themes to opt out or to provide a set aspect ratio that's better suited for the theme design. I'm curious to hear from @WordPress/gutenberg-design and @WordPress/block-themers around this too as this would change the default of how the site logo block would work. This would still leave the aspect ratio/cropping tools for users to change back and theme authors could override the default.

**Comment by scruffian** [comment]
This isn't trivial. The Site Logo block doesn't do any cropping on its own, it uses the WordPress tools to achieve this and when it saves the cropped image it syncs it back to the site icon. We could apply some cropping on the block via CSS but then this wouldn't apply to the site icon, so I would advise against it.

**Comment by jasmussen** [comment]
This would not be a hard crop, it would be a soft-crop, same as what the Aspect Ratio tool does for the Image block. So what I'm reading here, and agree with, is a desired behavior of new Site Logo blocks being inserted defaulting to have a 1:1 square aspect ratio applied to the placeholder. Further, that this aspect ratio, as well as width and height, persist even when you add an image, or add a different image, regardless of those other images aspect ratios or dimensions. 

Here's a mockup that advances the site logo a little bit in that regard:

<img width="904" alt="Site logo w  aspect ratio" src="https://github.com/WordPress/gutenberg/assets/1204802/63809880-e5f4-4806-bf63-db7506f5871e">

<img width="904" alt="Site logo w  aspect ratio, and image" src="https://github.com/WordPress/gutenberg/assets/1204802/5413650b-d8a6-40e3-bdb2-9b4882b635f7">

New in this mockup, compared to trunk:

* There's no button in the placeholder itself, only "Add media" in the toolbar, based on [this feedback](https://github.com/WordPress/gutenberg/issues/63327#issuecomment-2220221448). 
* "Add media" is, as mentioned, added to the toolbar.
* Dimensions—width, height, aspect ratios—are shown in the inspector, in place of the "Image width" that's there currently. Note that these dimensions are shown both for placeholder and final states.
* Resize handles are added.

These mockups show a slightly longer view, and we can start with less, such as simply adding the aspect ratio control, and making it persist between placeholder and final states.

**Comment by richtabor** [comment]
Should the add control be "Add logo" to be a bit more apparent?

**Comment by jasmussen** [comment]
Good point. I also missed updating the block toolbar when one is set. Here are fresh ones:

<img width="904" alt="Site logo w  aspect ratio" src="https://github.com/WordPress/gutenberg/assets/1204802/3d1f31cd-45f4-47aa-b083-9630dc671c29">

<img width="904" alt="Site logo w  aspect ratio, and image" src="https://github.com/WordPress/gutenberg/assets/1204802/df82d0e6-6a5e-46e6-a2fe-d181627bc8ee">

[Figma](https://www.figma.com/design/f4rGFqRCB2x0v4yo28dSUN/Placeholders?node-id=18-2284&t=uTHup1T9i5QvYwFD-1).

**Comment by carolinan** [comment]
Related: https://github.com/WordPress/gutenberg/issues/40838

**Comment by luminuu** [comment]
I'd rather make this opt-in instead of forcing themes to opt-out. This will be annoying for existing themes if they're used on a site with a site logo set that isn't 1:1, that would suddenly become cropped or squished together. It might also be confusing for users why suddenly the logo is not working as expected, if they are used to the image being displayed with the original aspect ratio.

**Comment by jasmussen** [comment]
The way I'm understanding "opt-out" is this: 

* If you are building a template and insert a site logo block, it defaults to a 1:1 aspect ratio applied.
* If you select this site logo block and set it to have its original aspect ratio, then save your template, your theme has opted out.

To that end, this is more of the default Site Logo block experience, than any separate theme opt-outs.

**Comment by richtabor** [comment]
> To that end, this is more of the default Site Logo block experience, than any separate theme opt-outs.

That's my thinking as well.

**Comment by Sam-Xronn** [comment]
Isn't it time to remove the site logo block now?

It's purpose has been super seeded by recent updates to the site editor. I understand why it was initially added, but it continues to cause so many more issues than it solves, I'm unsure why we're continue to push this block.

**Comment by ndiego** [comment]
> It's purpose has been super seeded by recent updates to the site editor. I understand why it was initially added, but it continues to cause so many more issues than it solves, I'm unsure why we're continue to push this block.

The Site Logo block is commonly used in most block themes, often in Header template parts. For those that don't need this block, it can [easily be disabled](https://developer.wordpress.org/news/2024/01/29/how-to-disable-specific-blocks-in-wordpress/).

**Comment by Sam-Xronn** [comment]
> > It's purpose has been super seeded by recent updates to the site editor. I understand why it was initially added, but it continues to cause so many more issues than it solves, I'm unsure why we're continue to push this block.
> 
> The Site Logo block is commonly used in most block themes, often in Header template parts. For those that don't need this block, it can [easily be disabled](https://developer.wordpress.org/news/2024/01/29/how-to-disable-specific-blocks-in-wordpress/).

It's commonly used due to it's early addition to the block editor not due to how useful it is. It no longer offers any apparent benefits compare to the image block. My concern is the continued waste of time diluting blocks into a level granularity not required. For example...

- Logo block doesn't support custom links, although a design system existing for this (image block). Making it useless to anyone running WP on a subdomain e.g. jobs. events. store. blog. etc... 
- Favicon support added to settings >general making it redundant (it's only real benefit)
- No support for multiple headers e.g. dark and light logos
- No aspect ratio support

It's not about showing/hiding the block and more about a confusing and poor UX. Of course you'd select the logo block to use as your logo if it exists, only to find out it offers no value and remove it and add an image block. Huh!

**Comment by ndiego** [comment]
> - Logo block doesn't support custom links, although a design system existing for this (image block). Making it useless to anyone running WP on a subdomain e.g. jobs. events. store. blog. etc...
> - Favicon support added to settings >general making it redundant (it's only real benefit)
> - No support for multiple headers e.g. dark and light logos
> - No aspect ratio support

Fair points. I wonder if extending the Image block to include a few new controls, like "Use as site logo" and deprecating the Site Logo block, would be a possible approach. Backward compatibility might be a nightmare, but we could then introduce an Image block variation called Site Logo so it's easy to find in the block inserter. This would provide the benefits of having a dedicated "block" for the Site Logo, but it would be powered by the Image block. 🤔

**Comment by Sam-Xronn** [comment]
> Backward compatibility might be a nightmare, but we could then introduce an Image block variation called Site Logo so it's easy to find in the block inserter.

This was a concern of mine, how possible the removal of a core block would be. Not sure if you could tie into the JSON versioning.

I'm still confused what a "Use site logo" option would change? As far as I'm aware the site logo doesn't change anything apart from offer an optional link to site homepage link which you can apply on an image. Or the ability to use as the site icon which should is now controlled in Settings > General. 

To avoid hijacking this thread further I wonder if we could have a wider discussion on this topic in its own issue?

**Comment by richtabor** [comment]
> Fair points. I wonder if extending the Image block to include a few new controls, like "Use as site logo" and deprecating the Site Logo block, would be a possible approach. Backward compatibility might be a nightmare, but we could then introduce an Image block variation called Site Logo so it's easy to find in the block inserter. This would provide the benefits of having a dedicated "block" for the Site Logo, but it would be powered by the Image block. 

As I mentioned https://github.com/WordPress/gutenberg/issues/63954#issuecomment-2253163170, sure we can configure an image block to be the logo, home url, etc, but it will always be more complicated than the Site Logo block, which you can add with one click, with much more limited control than an image block. But if a user wants to use an image block for a site logo, that's also just as fine—accepting the limitations of that route all the same. 

But let's keep this issue focused on the aspect ratio control being applied to the Site Logo block.
